src/components/ingredients-list.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Button from '../components/button';
+import RecipeIngredient from '../components/recipe-ingredient';
+import RecipeIngredientAmount from '../models/recipe-ingredient-amount';
+import styles from './ingredients-list.module.css';
+
+export default class IngredientsList extends React.Component {
+  static propTypes = {
+    className: PropTypes.string.isRequired,
+    ingredients: PropTypes.arrayOf(PropTypes.shape({
+      amount: PropTypes.oneOfType([
```

Comment:
If there was special rules around the two other than how to render them, I think this would be more appropriate. Not sure if this feels like two different object types here to me